tools/crm_attribute.c

```diff
@@ -790,6 +821,14 @@ main(int argc, char **argv)
         options.attr_options |= pcmk__node_attr_utilization;
     }
 
+    if (pcmk_any_flags_set(options.attr_options, pcmk__node_attr_sync_cluster|pcmk__node_attr_sync_local) &&
+        !try_ipc_update()) {
+        exit_code = CRM_EX_USAGE;
+        g_set_error(&error, PCMK__EXITC_ERROR, exit_code,
+                    "Error: --wait can only be used with --lifetime=reboot and either delete or update");
+        goto done;
+    }
```

Comment:
Hm, true.  This adds sync point support for those cases where we're talking to attrd for transient attributes.  Is there a better way to make all these options make sense?

Reply:
I've removed the block checking what options --wait was used with.

Without other changes, --wait=cluster --lifetime=forever shouldn't actaully be a problem.  The sync point flag (pcmk__node_attr_sync_*) only gets used if we're talking to attrd, and we're only talking to attrd if options.type is "reboot" (see use_attrd and set_type).

Reply:
> Without other changes, --wait=cluster --lifetime=forever shouldn't actaully be a problem.

The problem is that the equivalent of --wait=cluster when the live CIB is being directly modified would be for the CIB change to be propagated to all nodes in the cluster. I believe cib_sync_call only waits for the local copy of the CIB to be effective, but I'm not 100% sure of that. It's not an issue for CIB_shadow/CIB_file since there's only a local copy in that case.